include/nodejs/node_osrm_support.hpp

```diff
@@ -1220,14 +1220,66 @@ argumentsToMatchParameter(const Nan::FunctionCallbackInfo<v8::Value> &args,
         params->tidy = tidy->BooleanValue();
     }
 
-    bool parsedSuccessfully = parseCommonParameters(obj, params);
-    if (!parsedSuccessfully)
+    if (obj->Has(Nan::New("waypoints").ToLocalChecked()))
     {
-        return match_parameters_ptr();
+        v8::Local<v8::Value> waypoints = obj->Get(Nan::New("waypoints").ToLocalChecked());
+        if (waypoints.IsEmpty())
+            return match_parameters_ptr();
+
+        // must be array
+        if (!waypoints->IsArray())
+        {
+            Nan::ThrowError(
+                "Waypoints must be an array of integers corresponding to the input coordinates.");
+            return false;
+        }
+
+        auto waypoints_array = v8::Local<v8::Array>::Cast(waypoints);
+        // must have at least two elements
+        if (waypoints_array->Length() < 2)
+        {
+            Nan::ThrowError("At least two waypoints must be provided");
+            return false;
+        }
+        auto coords_size = params->coordinates.size();
+        auto waypoints_array_size = waypoints_array->Length();
+        for (uint32_t i = 0; i < waypoints_array_size; ++i)
```

Comment:
should be here `for (uint32_t i = 1; i < waypoints_array_size - 1; ++i)` as the first and the last waypoints are checked  separately?